rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -336,16 +624,23 @@ NodeParameters::get_parameter_types(const std::vector<std::string> & names) cons
 {
   std::lock_guard<std::mutex> lock(mutex_);
   std::vector<uint8_t> results;
```

Comment:
@wjwwood nit: consider doing `results.reserve(names.size())` to avoid incremental reallocations below.